make/migrations/postgresql/0030_1.11.0_schema.up.sql

```diff
@@ -0,0 +1,37 @@
+/* TODO remove the table artifact_2 and use the artifact instead after finishing the upgrade work */
+CREATE TABLE artifact_2
+(
+  id            SERIAL PRIMARY KEY NOT NULL,
+  /* image, chart, etc */
+  type          varchar(255),
+  project_id    int NOT NULL,
+  repository_id int NOT NULL,
+  media_type    varchar(255),
+  digest        varchar(255) NOT NULL,
+  size          int,  
+  upload_time   timestamp default CURRENT_TIMESTAMP,
+  extra_attrs   text,
+  annotations   text,
+  CONSTRAINT unique_artifact_2 UNIQUE (repository_id, digest)
+);
+
+CREATE TABLE tag
+(
+  id            SERIAL PRIMARY KEY NOT NULL,
+  repository_id int NOT NULL,
+  artifact_id   int NOT NULL,
+  name          varchar(255),
+  upload_time   timestamp default CURRENT_TIMESTAMP,
+  latest_download_time timestamp,
+  CONSTRAINT unique_tag UNIQUE (repository_id, name)
+);
+
+/* reference records the child artifact referenced by parent artifact */
+CREATE TABLE reference
```

Comment:
how about to name it as artifact_reference

Reply:
Updated